compiler/passes/docs.cpp

```diff
@@ -513,37 +519,32 @@ std::string generateSphinxProject(std::string dirpath) {
   //        provided... (thomasvandoren, 2015-01-29)
 
   // Create the output dir under the docs output dir.
-  std::string htmldir = dirpath + std::string("/html");
+  const char * htmldir = astr(dirpath.c_str(), "/html");
 
   // Ensure output directory exists.
-  std::string mkdirCmd = std::string("mkdir -p ") + htmldir;
-  mysystem(mkdirCmd.c_str(), "creating docs output dir");
+  const char * mkdirCmd = astr("mkdir -p ", htmldir);
+  mysystem(mkdirCmd, "creating docs output dir");
 
   // Copy the sphinx template into the output dir.
-  std::string sphinxTemplate = std::string(CHPL_HOME) + std::string("/third-party/chpldoc-venv/chpldoc-sphinx-project/*");
-  std::string cmd = std::string("cp -r ") + sphinxTemplate + std::string(" ") + htmldir + "/";
-  mysystem(cmd.c_str(), "copying chpldoc sphinx template");
+  const char * sphinxTemplate = astr(CHPL_HOME, "/third-party/chpldoc-venv/chpldoc-sphinx-project/*");
+  const char * cmd = astr("cp -r ", sphinxTemplate, " ", htmldir, "/");
+  mysystem(cmd, "copying chpldoc sphinx template");
 
-  std::string moddir = htmldir + "/source/modules";
-  return moddir;
+  const char * moddir = astr(htmldir, "/source/modules");
+  return std::string(moddir);
 }
 
 /* Call `make html` from inside sphinx project. */
 void generateSphinxOutput(std::string dirpath) {
-  std::string htmldir = dirpath + std::string("/html");
+  const char * htmldir = astr(dirpath.c_str(), "/html");
 
   // The virtualenv activate script is at:
   //   $CHPL_HOME/third-party/chpldoc-venv/install/$CHPL_TARGET_PLATFORM/chpldoc-virtualenv/bin/activate
-  std::string activate =
-    CHPL_HOME +
-    std::string("/third-party/chpldoc-venv/install/") +
-    CHPL_TARGET_PLATFORM +
-    std::string("/chpdoc-virtualenv/bin/activate");
+  const char * activate = astr(CHPL_HOME, "/third-party/chpldoc-venv/install/", CHPL_TARGET_PLATFORM, "/chpdoc-virtualenv/bin/activate");
```

Comment:
For readability, you may want to separate the call arguments onto separate lines.